src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/Target_ARM/ARMEmitter.cs

```diff
@@ -18,12 +18,41 @@ public ARMEmitter(NodeFactory factory, bool relocsOnly)
         public ObjectDataBuilder Builder;
         public TargetRegisterMap TargetRegister;
 
+        // mov reg, immediate
+        // reg range: 0-7
```

Comment:
Of course, I was going to do it later. Need to replace arguments with byte, short on int everywhere, so that there is no type conversions and it controlled the validity completely. By the way, I'm surprised that no one has done such a code for x64.